drivers/rtc/rtc_ll_stm32.c

```diff
@@ -0,0 +1,172 @@
+		LOG_ERR("clock configure failed\n");
+		return -EIO;
+	}
+
+	LL_RCC_EnableRTC();
+
+	z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
+
+	if (LL_RTC_Init(RTC, ((LL_RTC_InitTypeDef *)&cfg->ll_rtc_config)) != SUCCESS) {
+		return -EIO;
+	}
+
+#ifdef RTC_CR_BYPSHAD
+	LL_RTC_DisableWriteProtection(RTC);
+	LL_RTC_EnableShadowRegBypass(RTC);
+	LL_RTC_EnableWriteProtection(RTC);
+#endif /* RTC_CR_BYPSHAD */
+
+	return 0;
+}
+
+static const struct stm32_pclken rtc_clk[] = STM32_DT_INST_CLOCKS(0);
+
+static const struct rtc_stm32_config rtc_config = {
+	.ll_rtc_config = {
+			.HourFormat = LL_RTC_HOURFORMAT_24HOUR,
+#if DT_INST_CLOCKS_CELL(1, bus) == STM32_SRC_LSI
+			/* prescaler values for LSI @ 32 KHz */
+			.AsynchPrescaler = 0x7F,
+			.SynchPrescaler = 0x00F9,
+#else /* DT_INST_CLOCKS_CELL(1, bus) == STM32_SRC_LSE */
+			/* prescaler values for LSE @ 32768 Hz */
+			.AsynchPrescaler = 0x7F,
+			.SynchPrescaler = 0x00FF,
+#endif
+		},
+	.pclken = rtc_clk,
+};
+
+static int rtc_stm32_set_time(const struct device *dev, const struct rtc_time *timeptr)
+{
+	LOG_INF("Setting clock");
+	LL_RTC_DisableWriteProtection(RTC);
+
+	LL_RTC_EnableInitMode(RTC);
+
+	while (!LL_RTC_IsActiveFlag_INIT(RTC)) {
+	};
+
+	LL_RTC_DATE_SetYear(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_year - TM_TO_RTC_OFFSET));
+	LL_RTC_DATE_SetMonth(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_mon + 1));
+	LL_RTC_DATE_SetDay(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_mday));
+
+	LL_RTC_TIME_SetHour(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_hour));
+	LL_RTC_TIME_SetMinute(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_min));
+	LL_RTC_TIME_SetSecond(RTC, __LL_RTC_CONVERT_BIN2BCD(timeptr->tm_sec));
+
+	LL_RTC_DisableInitMode(RTC);
+
+	LL_RTC_EnableWriteProtection(RTC);
+
+	return 0;
+}
+
+static int rtc_stm32_get_time(const struct device *dev, struct rtc_time *timeptr)
+{
+	ARG_UNUSED(dev);
+
+	uint32_t rtc_date, rtc_time;
+
+	/* Read time and date registers */
+	rtc_time = LL_RTC_TIME_Get(RTC);
+	rtc_date = LL_RTC_DATE_Get(RTC);
+
+	timeptr->tm_year = TM_TO_RTC_OFFSET + __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_YEAR(rtc_date));
+	/* tm_mon allowed values are 0-11 */
+	timeptr->tm_mon = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_MONTH(rtc_date)) - 1;
+	timeptr->tm_mday = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_DAY(rtc_date));
+
+	timeptr->tm_hour = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_HOUR(rtc_time));
+	timeptr->tm_min = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_MINUTE(rtc_time));
+	timeptr->tm_sec = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_SECOND(rtc_time));
+
+	timeptr->tm_nsec = LL_RTC_TIME_GetSubSecond(RTC);
+
+	return 0;
+}
+
+struct rtc_driver_api rtc_stm32_driver_api = {
+	.set_time = rtc_stm32_set_time,
+	.get_time = rtc_stm32_get_time,
+#if defined(CONFIG_RTC_ALARM)
+#error Alarm not supported yet for this RTC
+#endif /* CONFIG_RTC_ALARM */
+
+#if defined(CONFIG_RTC_UPDATE)
+#error RTC_UPDATE not supported yet for this RTC
+#endif /* CONFIG_RTC_UPDATE */
```

Comment:
API will return -ENOSYS if you don't treat these cases.
There is no need to generate errors if enabled in the driver.
Also, if you have another RTC component on your board, this will prevent to use this functionality, even if supported by the other driver.

Reply:
OK - will remove #error lines